src/cluster/placement/storage/storage.go

```diff
@@ -50,98 +52,93 @@ func NewPlacementStorage(store kv.Store, key string, opts placement.Options) pla
 	}
 }
 
-func (s *storage) CheckAndSetProto(p proto.Message, version int) error {
+func (s *storage) CheckAndSetProto(p proto.Message, version int) (int, error) {
 	if err := s.helper.ValidateProto(p); err != nil {
-		return err
+		return errorVersionValue, err
 	}
 
 	if s.opts.Dryrun() {
 		s.logger.Info("this is a dryrun, the operation is not persisted")
-		return nil
+		return errorVersionValue, nil
```

Comment:
nit: probably not ideal to apply the `+1` logic here but this should be `version+1` for dryrun?

Reply:
Ehh, I'd rather it be a constant for dry runs. `version+1` is only the case because that's an implementation detail of etcd. Plus between when you do a dry run and the actual command someone else could've changed the placement.

Reply:
It just feels weird to return an `errVersionValue` for this case when it's not an error, but I guess that's probably fine since this is just dryrun, the version would show up in logs after you run m3admin though, it does reduce readability of the log somewhat.